lib/index.js

```diff
@@ -79,30 +83,48 @@ var redirectCodes = new Set([301, 302, 303, 307]);
  * @param {String} url
  * @param {Object} options
  * @param {Number} tryCount Prevent infinite redirects.
+ * @param {Object} reconnectInfo Continue after ECONNRESET
  */
-function doDownload(stream, url, options, tryCount) {
+function doDownload(stream, url, options, tryCount, reconnectInfo) {
   if (stream._isDestroyed) { return; }
   if (tryCount === 0) {
     stream.emit('error', new Error('Too many redirects'));
     return;
   }
+  if (reconnectInfo.trys === 0) {
+    stream.emit('error', new Error('Too many reconnects'));
+    return;
+  }
 
   // Start downloading the video.
   var myrequest = options.request || request;
-  var req = myrequest(url, options.requestOptions);
+  var rangedUrl = url;
+  if (reconnectInfo.downloaded !== 0 ||
+      reconnectInfo.range.start !== 0 ||
+      reconnectInfo.range.end !== -1) {
+    var start = reconnectInfo.range.start + reconnectInfo.downloaded;
+    var end = reconnectInfo.range.end != -1 ? reconnectInfo.range.end : '';
+    rangedUrl += '&range=' + start + '-' + end;
+  }
+  var req = myrequest(rangedUrl, options.requestOptions);
 
   var myres;
   stream.destroy = function() {
+    stream._isDestroyed = true;
     req.abort();
     if (myres) {
-      myres.destroy();
       myres.unpipe();
+      myres.destroy();
     }
     stream.emit('abort');
   };
 
   req.on('error', function(err) {
-    stream.emit('error', err);
+    if (stream._isDestroyed || !myres) {
+      stream.emit('error', err);
```

Comment:
Not sure wether u shouldnt close the stream anyway but since i don't know better...